trl/trainer/online_dpo_trainer.py

```diff
@@ -162,6 +167,16 @@ def __init__(
             "val/contain_eos_token": [],
         }
 
+        self.generation_config = GenerationConfig(
+            max_new_tokens=args.max_new_tokens,
+            min_new_tokens=args.max_new_tokens,
+            temperature=args.temperature,
+            top_k=0,
+            top_p=1.0,
+            do_sample=True,
+            use_cache=False if args.gradient_checkpointing else True,
+        )
```

Comment:
Out of curiosity, why do you need this change?

Reply:
i realized we were making this for every step and since its fixed for now... i moved it over to the main __init__

Reply:
ok I got it, so that you don't have to define it again in XPOTrainer

Reply:
yes that too haha